cip25/rust/src/utils.rs

```diff
@@ -109,6 +109,87 @@ impl From<&ChunkableString> for String {
     }
 }
 
+
+/// A subset of MetadataDetails where the keys are optional
+/// Useful to extract the key fields (name & image) of incorrectly formatted cip25
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, schemars::JsonSchema)]
+pub struct MiniMetadataDetails {
+    pub name: Option<String64>,
+    pub image: Option<ChunkableString>,
+}
+
+impl MiniMetadataDetails {
+    pub fn new(name: Option<String64>, image: Option<ChunkableString>) -> Self {
+        Self {
+            name,
+            image,
+        }
+    }
+
+    /// loose parsing of CIP25 metadata to allow for common exceptions to the format
+    /// TODO: this is not an ideal solution
+    ///       ideally: we would have a function that takes in a policy ID
+    ///       and would have a lookup map to know which lambda to call to get the name & image depending on the policy ID
+    ///       with a fallback to the standard CIP25 definition
+    ///       however, since this is a lot of work, we use this temporary solution instead
+    pub fn loose_parse(data: Vec<u8>) -> Result<Self, DeserializeError> {
+        let metadatum = TransactionMetadatum::from_bytes(
+            data,
+        )?;
+        match metadatum {
+            TransactionMetadatum::Map { entries, .. } => {
+                let name: Option<String64> = entries
+                .get(&TransactionMetadatum::new_text("name".to_owned()))
+                .or_else(|| entries.get(&TransactionMetadatum::new_text("name".to_owned())))
```

Comment:
This is a duplicate (`name` a second time)

Reply:
And will be removed before the PR is done, right?